src/test/java/com/google/api/codegen/testdata/nodejs_test_library.baseline

```diff
@@ -51,6 +54,29 @@ describe('LibraryServiceClient', function() {
         done();
       });
     });
+
+    it('invokes createShelf with error', function(done) {
+      var client = libraryV1.libraryServiceClient();
+      // Mock request
+      var shelf = {};
+      var request = {
+          shelf : shelf
+      };
+
+      // Mock Grpc layer
+      client._createShelf = function(actualRequest, options, callback) {
```

Comment:
I think the creation of the mock method can be consolidated to reduce code reuse as seen below:

Utility method used by non-streaming tests:
```js
function mockGrpcMethod(expectedRequest, response) {
  return function(actualRequest, options, callback) {
    assert.deepStrictEqual(actualRequest, expectedRequest);
    if (response instanceof Error) { // Unsure if idiomatic.
      callback(response);
    } else {
      callback(null, response);
    }
  };
}
```

When mocking the grpc layer for a regular test:
```js
// Non-error testing
client._getBook = mockGrpcMethod(expectedRequest, expectedResponse);
```

And for an error:
```js
// This will only need to be declared once.
var error = new Error('Fake error');
error.code = FAKE_STATUS_CODE;

// Error testing
client._getBook = mockGrpcMethod(expectedRequest, error);
```

Something similar can be made for streaming as well.

Note: Theres a line in my example that I am unsure if it is idiomatic. @jmuk can you take a quick look.